zerver/views/upload.py

```diff
@@ -120,7 +119,13 @@ def serve_local(
     if not os.path.isfile(local_path):
         return HttpResponseNotFound("<p>File not found</p>")
 
-    mimetype, encoding = guess_type(path_id)
+    try:
+        filename = Attachment.objects.get(path_id=path_id).file_name
+    except Attachment.DoesNotExist:
+        # Thumbnail paths do not exist in the Attachments table, and
+        # we should serve them with their name as, e.g., "100x75.webp"
+        filename = os.path.basename(path_id)
```

Comment:
This adds an extra database query in a hot code path; we could probably plumb it through from `validate_attachment_request`.